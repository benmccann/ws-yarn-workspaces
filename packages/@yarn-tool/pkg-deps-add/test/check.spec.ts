//@noUnusedParameters:false

import { basename, extname } from 'path';
import { checkDependenciesExistsAll } from '../index';
import { IPackageJson } from '@ts-type/package-dts/package-json';

beforeAll(async () =>
{

});

describe(basename(__filename, extname(__filename)), () =>
{
	let pkg: IPackageJson;

	beforeAll(async () =>
	{
		pkg = {
			dependencies: {
				"tslib": "^2"
			},
			devDependencies: {
				"tslib": "^2",
				"@types/node": "^18.7.13",
			},
			peerDependencies: {
				"tslib": "^2"
			},
			optionalDependencies: {
				"tslib": "^2"
			}
		}
	});

	test(`test`, () =>
	{
		expect(checkDependenciesExistsAll(pkg, "tslib")).toMatchSnapshot();
		expect(checkDependenciesExistsAll(pkg, "@types/node")).toMatchSnapshot();
	});

})
