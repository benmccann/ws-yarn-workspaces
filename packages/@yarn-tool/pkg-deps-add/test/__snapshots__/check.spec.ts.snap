// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check.spec test 1`] = `
{
  "_exists": true,
  "_field": [
    "dependencies",
    "devDependencies",
    "peerDependencies",
    "optionalDependencies",
  ],
  "dependencies": true,
  "devDependencies": true,
  "optionalDependencies": true,
  "peerDependencies": true,
}
`;

exports[`check.spec test 2`] = `
{
  "_exists": true,
  "_field": [
    "devDependencies",
  ],
  "dependencies": false,
  "devDependencies": true,
  "optionalDependencies": false,
  "peerDependencies": false,
}
`;
